config/skills.json

```diff
@@ -0,0 +1,1027 @@
+                }
+            ]
+        },
+        "expert" : {
+            "description" : "",
+            "effects" : [
+                {
+                    "subtype" : "skill.wisdom",
+                    "type" : "SECONDARY_SKILL_PREMY",
+                    "val" : 3,
+                    "valueType" : "BASE_NUMBER"
+                }
+            ]
+        }
+    },
+    "mysticism" : {
+        "basic" : {
+            "description" : "",
+            "effects" : [
+                {
+                    "type" : "MANA_REGENERATION",
+                    "val" : 1,
+                    "valueType" : "BASE_NUMBER"
+                }
+            ]
+        },
+        "advanced" : {
+            "description" : "",
+            "effects" : [
+                {
+                    "type" : "MANA_REGENERATION",
+                    "val" : 2,
+                    "valueType" : "BASE_NUMBER"
+                }
+            ]
+        },
+        "expert" : {
+            "description" : "",
+            "effects" : [
+                {
+                    "type" : "MANA_REGENERATION",
+                    "val" : 3,
+                    "valueType" : "BASE_NUMBER"
+                }
+            ]
+        }
+    },
+    "luck" : {
+        "basic" : {
+            "description" : "",
+            "effects" : [
+                {
+                    "type" : "LUCK",
+                    "val" : 1,
+                    "valueType" : "BASE_NUMBER"
+                }
+            ]
+        },
+        "advanced" : {
+            "description" : "",
+            "effects" : [
+                {
+                    "type" : "LUCK",
+                    "val" : 2,
+                    "valueType" : "BASE_NUMBER"
+                }
+            ]
+        },
+        "expert" : {
+            "description" : "",
+            "effects" : [
+                {
+                    "type" : "LUCK",
+                    "val" : 3,
+                    "valueType" : "BASE_NUMBER"
+                }
+            ]
+        }
+    },
+    "ballistics" : {
+        "basic" : {
+            "description" : "",
+            "effects" : [
+                {
+                    "subtype" : "skill.ballistics",
+                    "type" : "SECONDARY_SKILL_PREMY",
+                    "val" : 1,
+                    "valueType" : "BASE_NUMBER"
+                },
+                {
+                    "subtype" : "creature.catapult",
+                    "type" : "MANUAL_CONTROL",
+                    "val" : 100,
+                    "valueType" : "BASE_NUMBER"
+                }
+            ]
+        },
+        "advanced" : {
+            "description" : "",
+            "effects" : [
+                {
+                    "subtype" : "skill.ballistics",
+                    "type" : "SECONDARY_SKILL_PREMY",
+                    "val" : 2,
+                    "valueType" : "BASE_NUMBER"
+                }
```

Comment:
missing MANUAL_CONTROL for advanced and expert?

Reply:
Shouldn't be needed as the hero receives the bonuses from all levels (although bonuses of the same type don't stack, only the highest applies). Just fixed a related bug with re-application of secondary skill bonuses though.

Reply:
Hero should have bonuses only from current skill level. Secondary skill level could be changed even in absolute mode and in any direction. See `IGameEventCallback::changeSecSkill`

Reply:
included in "base" effects now